### ISSUE TITLE
x64-windows gdk-pixbuf build failure

### ISSUE THREAD

**Issue by youknowone** [microsoft/vcpkg#32254]
vcpkg version:  2023.06.20

build log

```
> ..\vcpkg\vcpkg.exe install gdk-pixbuf --triplet x64-windows --debug --editable
[DEBUG] To include the environment variables in debug output, pass --debug-env
[DEBUG] Disabling metrics because vcpkg.disable-metrics exists
[DEBUG] Trying to load bundleconfig from E:\vcpkg\vcpkg-bundle.json
[DEBUG] Failed to open: E:\vcpkg\vcpkg-bundle.json
[DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[DEBUG] Feature flag 'binarycaching' unset
[DEBUG] Feature flag 'compilertracking' unset
[DEBUG] Feature flag 'registries' unset
[DEBUG] Feature flag 'versions' unset
[DEBUG] Using scripts-root: E:\vcpkg\scripts
[DEBUG] Using builtin-ports: E:\vcpkg\ports
[DEBUG] Using installed-root: E:\vcpkg\installed
[DEBUG] Using buildtrees-root: E:\vcpkg\buildtrees
[DEBUG] Using packages-root: E:\vcpkg\packages
[DEBUG] Using vcpkg-root: E:\vcpkg
[DEBUG] Using scripts-root: E:\vcpkg\scripts
[DEBUG] Using builtin-registry: E:\vcpkg\versions
[DEBUG] Using downloads-root: E:\vcpkg\downloads
[DEBUG] Detecting host with IsWow64Process2
Computing installation plan...
[DEBUG] Loading dep info for: gdk-pixbuf:x64-windows gettext:x64-windows glib:x64-windows libpng:x64-windows vcpkg-tool-meson:x64-windows
A suitable version of cmake was not found (required v3.25.1) Downloading portable cmake 3.25.1...
Downloading cmake...
https://github.com/Kitware/CMake/releases/download/v3.25.1/cmake-3.25.1-windows-i386.zip->E:\vcpkg\downloads\cmake-3.25.1-windows-i386.zip
Downloading https://github.com/Kitware/CMake/releases/download/v3.25.1/cmake-3.25.1-windows-i386.zip
[DEBUG] Trying to hash E:\vcpkg\downloads\cmake-3.25.1-windows-i386.zip.17656.part
[DEBUG] E:\vcpkg\downloads\cmake-3.25.1-windows-i386.zip.17656.part has hash 4a84abdb53781178a99046aa09cf6a6b2bfe159d6b6bbdd2449839d2f50b872b4cc783c44774343b6122ec9400bcfdec65fffd4a246575ba55d178bc180f6adc
Extracting cmake...
[DEBUG] CreateProcessW("C:\WINDOWS\system32\tar.exe" xzf "E:\vcpkg\downloads\cmake-3.25.1-windows-i386.zip")
[DEBUG] cmd_execute() returned 0 after 7137416 us
[DEBUG] CreateProcessW("E:\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe" --version)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1000: cmd_execute_and_stream_data() returned 0 after   141111 us
[DEBUG] CreateProcessW("E:\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe" -DVCPKG_ROOT_DIR=E:/vcpkg -DPACKAGES_DIR=E:/vcpkg/packages -DBUILDTREES_DIR=E:/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=E:/vcpkg/installed -DDOWNLOADS=E:/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "E:\vcpkg\buildtrees\0.vcpkg_dep_info.cmake")
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1001: cmd_execute_and_stream_data() returned 0 after    63988 us
[DEBUG] CreateProcessW("E:\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe" -DVCPKG_ROOT_DIR=E:/vcpkg -DPACKAGES_DIR=E:/vcpkg/packages -DBUILDTREES_DIR=E:/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=E:/vcpkg/installed -DDOWNLOADS=E:/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "E:\vcpkg\buildtrees\0.vcpkg_tags.cmake")
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1002: cmd_execute_and_stream_data() returned 0 after    62128 us
The following packages will be built and installed:
    gdk-pixbuf:x64-windows -> 2.42.10#1
[DEBUG] CreateProcessW("C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe" -all -prerelease -legacy -products * -format xml)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1003: cmd_execute_and_stream_data() returned 0 after    55962 us
[DEBUG] Binary caching for package gdk-pixbuf:x64-windows is disabled due to --editable
[DEBUG] Default binary cache path is: C:\Users\user\AppData\Local\vcpkg\archives
Installing 1/1 gdk-pixbuf:x64-windows...
Building gdk-pixbuf:x64-windows...
[DEBUG] Found path: C:\Program Files\Git\cmd\git.exe
[DEBUG] CreateProcessW("C:\Program Files\Git\cmd\git.exe" --version)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1004: cmd_execute_and_stream_data() returned 0 after    68524 us
[DEBUG] Detecting host with IsWow64Process2
[DEBUG] Found path: C:\Program Files\PowerShell\7\pwsh.exe
[DEBUG] CreateProcessW("C:\Program Files\PowerShell\7\pwsh.exe" --version)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1005: cmd_execute_and_stream_data() returned 0 after   306821 us
[DEBUG] command line: cmd /c "C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Auxiliary/Build\vcvarsall.bat" -vcvars_ver=14.35.32215 amd64   2>&1 <NUL  & echo cdARN4xjKueKScMy9C6H & set
[DEBUG] CreateProcessW(cmd /c "C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Auxiliary/Build\vcvarsall.bat" -vcvars_ver=14.35.32215 amd64   2>&1 <NUL  & echo cdARN4xjKueKScMy9C6H & set)
[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1006: cmd_execute_and_stream_data() returned 0 after  2347918 us
[DEBUG] **********************************************************************
** Visual Studio 2022 Developer Command Prompt v17.5.5
** Copyright (c) 2022 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
cdARN4xjKueKScMy9C6H
ALLUSERSPROFILE=C:\ProgramData
APPDATA=C:\Users\user\AppData\Roaming
CommandPromptType=Native
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
ComSpec=C:\WINDOWS\system32\cmd.exe
DevEnvDir=C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\
ExtensionSdkDir=C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs
EXTERNAL_INCLUDE=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\ATLMFC\include;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Auxiliary\VS\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\shared;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um
Framework40Version=v4.0
FrameworkDir=C:\Windows\Microsoft.NET\Framework64\
FrameworkDir64=C:\Windows\Microsoft.NET\Framework64\
FrameworkVersion=v4.0.30319
FrameworkVersion64=v4.0.30319
HOMEDRIVE=C:
HOMEPATH=\Users\user
HTMLHelpDir=C:\Program Files (x86)\HTML Help Workshop
INCLUDE=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\ATLMFC\include;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Auxiliary\VS\include;C:\Program Files (x86)\Windows Kits\10\include\10.0.19041.0\ucrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\um;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\shared;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\winrt;C:\Program Files (x86)\Windows Kits\10\\include\10.0.19041.0\\cppwinrt;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um
LIB=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\ATLMFC\lib\x64;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\lib\x64;C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64;C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\ucrt\x64;C:\Program Files (x86)\Windows Kits\10\\lib\10.0.19041.0\\um\x64
LIBPATH=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\ATLMFC\lib\x64;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\lib\x64;C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\lib\x86\store\references;C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.19041.0;C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0;C:\Windows\Microsoft.NET\Framework64\v4.0.30319
LOCALAPPDATA=C:\Users\user\AppData\Local
NETFXSDKDir=C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\
NUMBER_OF_PROCESSORS=8
OS=Windows_NT
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL
Platform=x64
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 158 Stepping 9, GenuineIntel
PROCESSOR_LEVEL=6
PROCESSOR_REVISION=9e09
ProgramData=C:\ProgramData
ProgramFiles=C:\Program Files
ProgramFiles(x86)=C:\Program Files (x86)
ProgramW6432=C:\Program Files
PROMPT=$P$G
PSModulePath=C:\Users\user\OneDrive\문서\PowerShell\Modules;C:\Program Files\PowerShell\Modules;c:\program files\powershell\7\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules;C:\Program Files (x86)\Windows Kits\10\Microsoft Application Virtualization\Sequencer\AppvPkgConverter;C:\Program Files (x86)\Windows Kits\10\Microsoft Application Virtualization\Sequencer\AppvSequencer;C:\Program Files (x86)\Windows Kits\10\Microsoft Application Virtualization\
PUBLIC=C:\Users\Public
SystemDrive=C:
SystemRoot=C:\WINDOWS
TEMP=C:\Users\user\AppData\Local\Temp
TMP=C:\Users\user\AppData\Local\Temp
UCRTVersion=10.0.19041.0
UniversalCRTSdkDir=C:\Program Files (x86)\Windows Kits\10\
USERNAME=USER
USERPROFILE=C:\Users\user
VCIDEInstallDir=C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\VC\
VCINSTALLDIR=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\
VCToolsInstallDir=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\
VCToolsRedistDir=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Redist\MSVC\14.34.31931\
VCToolsVersion=14.35.32215
VisualStudioVersion=17.0
VS170COMNTOOLS=C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\Tools\
VSCMD_ARG_app_plat=Desktop
VSCMD_ARG_HOST_ARCH=x64
VSCMD_ARG_TGT_ARCH=x64
VSCMD_ARG_VCVARS_VER=14.35.32215
VSCMD_VER=17.5.5
VSINSTALLDIR=C:\Program Files\Microsoft Visual Studio\2022\Professional\
windir=C:\WINDOWS
CUDA_PATH=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
CUDA_PATH_V10_1=C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
NVCUDASAMPLES_ROOT=C:\ProgramData\NVIDIA Corporation\CUDA Samples\v10.1
NVTOOLSEXT_PATH=C:\Program Files\NVIDIA Corporation\NvToolsExt\
PATH=C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64;C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\VC\VCPackages;C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Current\bin\Roslyn;C:\Program Files\Microsoft Visual Studio\2022\Professional\Team Tools\Performance Tools\x64;C:\Program Files\Microsoft Visual Studio\2022\Professional\Team Tools\Performance Tools;C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\;C:\Program Files (x86)\HTML Help Workshop;C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\\x64;C:\Program Files (x86)\Windows Kits\10\bin\\x64;C:\Program Files\Microsoft Visual Studio\2022\Professional\\MSBuild\Current\Bin\amd64;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\;C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\Tools\;C:\Program Files\PowerShell\7;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\system32\Wbem;C:\WINDOWS\system32\WindowsPowerShell\v1.0\;C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\VC\Linux\bin\ConnectionManagerExe
VSLANG=1033
VSCMD_SKIP_SENDTELEMETRY=1
VCPKG_COMMAND=E:/vcpkg/vcpkg.exe
WindowsLibPath=C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.19041.0;C:\Program Files (x86)\Windows Kits\10\References\10.0.19041.0
WindowsSdkBinPath=C:\Program Files (x86)\Windows Kits\10\bin\
WindowsSdkDir=C:\Program Files (x86)\Windows Kits\10\
WindowsSDKLibVersion=10.0.19041.0\
WindowsSdkVerBinPath=C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\
WindowsSDKVersion=10.0.19041.0\
WindowsSDK_ExecutablePath_x64=C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\x64\
WindowsSDK_ExecutablePath_x86=C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\
X_VCPKG_RECURSIVE_DATA={
"VCPKG_ROOT_ENV": "E:\\vcpkg"
}

__DOTNET_ADD_64BIT=1
__DOTNET_PREFERRED_BITNESS=64
__VSCMD_PREINIT_PATH=C:\Program Files\PowerShell\7;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\system32\Wbem;C:\WINDOWS\system32\WindowsPowerShell\v1.0\

[DEBUG] CreateProcessW("E:\vcpkg\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe" "-DALL_FEATURES=introspection;" -DCURRENT_PORT_DIR=E:/vcpkg/ports/gdk-pixbuf -D_HOST_TRIPLET=x64-windows -DFEATURES=core -DPORT=gdk-pixbuf -DVERSION=2.42.10 -DVCPKG_USE_HEAD_VERSION=0 -D_VCPKG_DOWNLOAD_TOOL=BUILT_IN -D_VCPKG_EDITABLE=1 -D_VCPKG_NO_DOWNLOADS=0 -DZ_VCPKG_CHAINLOAD_TOOLCHAIN_FILE=E:/vcpkg/scripts/toolchains/windows.cmake -DCMD=BUILD -DDOWNLOADS=E:/vcpkg/downloads -DTARGET_TRIPLET=x64-windows -DTARGET_TRIPLET_FILE=E:/vcpkg/triplets/x64-windows.cmake -DVCPKG_BASE_VERSION=2023-06-15 -DVCPKG_CONCURRENCY=9 -DVCPKG_PLATFORM_TOOLSET=v143 "-DGIT=C:/Program Files/Git/cmd/git.exe" "-DVCPKG_PORT_CONFIGS=E:\vcpkg\installed\x64-windows\share\gettext\vcpkg-port-config.cmake;E:\vcpkg\installed\x64-windows\share\vcpkg-tool-meson\vcpkg-port-config.cmake" -DVCPKG_ROOT_DIR=E:/vcpkg -DPACKAGES_DIR=E:/vcpkg/packages -DBUILDTREES_DIR=E:/vcpkg/buildtrees -D_VCPKG_INSTALLED_DIR=E:/vcpkg/installed -DDOWNLOADS=E:/vcpkg/downloads -DVCPKG_MANIFEST_INSTALL=OFF -P "E:\vcpkg\scripts\ports.cmake")
-- Using cached GNOME-gdk-pixbuf-2.42.10.tar.gz.
-- Using source at E:/vcpkg/buildtrees/gdk-pixbuf/src/2.42.10-ebb7e46440
-- Getting CMake variables for x64-windows
-- Configuring x64-windows-dbg
-- Using cached mingw-w64-i686-pkgconf-1~1.8.0-2-any.pkg.tar.zst.
-- Using cached msys2-msys2-runtime-3.4.6-1-x86_64.pkg.tar.zst.
-- Using msys root at E:/vcpkg/downloads/tools/msys2/6f3fa1a12ef85a6f
-- Configuring x64-windows-dbg done
-- Configuring x64-windows-rel
-- Configuring x64-windows-rel done
-- Package x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: E:/vcpkg/downloads/tools/ninja/1.10.2-windows/ninja.exe install -v
    Working Directory: E:/vcpkg/buildtrees/gdk-pixbuf/x64-windows-dbg
    Error code: 1
    See logs for more information:
      E:\vcpkg\buildtrees\gdk-pixbuf\package-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_meson.cmake:33 (vcpkg_execute_required_process)
  ports/gdk-pixbuf/portfile.cmake:61 (vcpkg_install_meson)
  scripts/ports.cmake:147 (include)


[DEBUG] ReadFile() finished with GetLastError(): 109
[DEBUG] 1007: cmd_execute_and_stream_data() returned 1 after 32504694 us
error: building gdk-pixbuf:x64-windows failed with: BUILD_FAILED
Elapsed time to handle gdk-pixbuf:x64-windows: 35 s
internal error: D:\a\_work\1\s\src\vcpkg\commands.build.cpp(1511): Value was null
Please open an issue at https://github.com/microsoft/vcpkg/issues/new?template=other-type-of-bug-report.md&labels=category:vcpkg-bug with detailed steps to reproduce the problem.
[DEBUG] D:\a\_work\1\s\src\vcpkg\commands.build.cpp(1511):
[DEBUG] Time in subprocesses: 42688562us
[DEBUG] Time in parsing JSON: 1693us
[DEBUG] Time in JSON reader: 1679us
[DEBUG] Time in filesystem: 17865us
[DEBUG] Time in loading ports: 20112us
[DEBUG] Exiting after 46 s (46358210us)
```

cmake log
```
...
[162/175] "C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe" "-Igdk-pixbuf\gdk-pixbuf-query-loaders.exe.p" "-Igdk-pixbuf" "-I..\src\2.42.10-ebb7e46440\gdk-pixbuf" "-I." "-I..\src\2.42.10-ebb7e46440" "-IE:/vcpkg/installed/x64-windows/debug/lib/pkgconfig//../../../include/glib-2.0" "-IE:/vcpkg/installed/x64-windows/debug/lib/pkgconfig//../../../include" "-IE:/vcpkg/installed/x64-windows/debug/lib/pkgconfig//../../lib/glib-2.0/include" "-IE:/vcpkg/installed/x64-windows/include" "/MDd" "/nologo" "/showIncludes" "/utf-8" "/W2" "/Od" "/Zi" "-D_POSIX_C_SOURCE=200809L" "-D_DEFAULT_SOURCE" "-D_XOPEN_SOURCE=700" "-utf-8" "-DGDK_PIXBUF_RELOCATABLE" "-DHAVE_CONFIG_H=1" "-nologo" "-DWIN32" "-D_WINDOWS" "-W3" "-utf-8" "-MP" "-D_DEBUG" "-MDd" "-Z7" "-Ob0" "-Od" "-RTC1" "-utf-8" "-DG_LOG_STRUCTURED=1" "-DG_LOG_DOMAIN=\"GdkPixbuf\"" "-DGDK_PIXBUF_COMPILATION" "-DGDK_PIXBUF_PREFIX=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug\"" "-DGDK_PIXBUF_LOCALEDIR=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug/share/locale\"" "-DGDK_PIXBUF_LIBDIR=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug/lib\"" "-DGDK_PIXBUF_BINARY_VERSION=\"2.10.0\"" "-DGDK_PIXBUF_ENABLE_BACKEND" "-DPIXBUF_LIBDIR=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug/lib/gdk-pixbuf-2.0/2.10.0/loaders\"" "-DBUILT_MODULES_DIR=\"E:\\vcpkg\\buildtrees\\gdk-pixbuf\\x64-windows-dbg\\gdk-pixbuf\"" "/Fdgdk-pixbuf\gdk-pixbuf-query-loaders.exe.p\queryloaders.c.pdb" /Fogdk-pixbuf/gdk-pixbuf-query-loaders.exe.p/queryloaders.c.obj "/c" ../src/2.42.10-ebb7e46440/gdk-pixbuf/queryloaders.c
FAILED: gdk-pixbuf/gdk-pixbuf-query-loaders.exe.p/queryloaders.c.obj 
"C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.35.32215/bin/Hostx64/x64/cl.exe" "-Igdk-pixbuf\gdk-pixbuf-query-loaders.exe.p" "-Igdk-pixbuf" "-I..\src\2.42.10-ebb7e46440\gdk-pixbuf" "-I." "-I..\src\2.42.10-ebb7e46440" "-IE:/vcpkg/installed/x64-windows/debug/lib/pkgconfig//../../../include/glib-2.0" "-IE:/vcpkg/installed/x64-windows/debug/lib/pkgconfig//../../../include" "-IE:/vcpkg/installed/x64-windows/debug/lib/pkgconfig//../../lib/glib-2.0/include" "-IE:/vcpkg/installed/x64-windows/include" "/MDd" "/nologo" "/showIncludes" "/utf-8" "/W2" "/Od" "/Zi" "-D_POSIX_C_SOURCE=200809L" "-D_DEFAULT_SOURCE" "-D_XOPEN_SOURCE=700" "-utf-8" "-DGDK_PIXBUF_RELOCATABLE" "-DHAVE_CONFIG_H=1" "-nologo" "-DWIN32" "-D_WINDOWS" "-W3" "-utf-8" "-MP" "-D_DEBUG" "-MDd" "-Z7" "-Ob0" "-Od" "-RTC1" "-utf-8" "-DG_LOG_STRUCTURED=1" "-DG_LOG_DOMAIN=\"GdkPixbuf\"" "-DGDK_PIXBUF_COMPILATION" "-DGDK_PIXBUF_PREFIX=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug\"" "-DGDK_PIXBUF_LOCALEDIR=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug/share/locale\"" "-DGDK_PIXBUF_LIBDIR=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug/lib\"" "-DGDK_PIXBUF_BINARY_VERSION=\"2.10.0\"" "-DGDK_PIXBUF_ENABLE_BACKEND" "-DPIXBUF_LIBDIR=\"E:/vcpkg/packages/gdk-pixbuf_x64-windows/debug/lib/gdk-pixbuf-2.0/2.10.0/loaders\"" "-DBUILT_MODULES_DIR=\"E:\\vcpkg\\buildtrees\\gdk-pixbuf\\x64-windows-dbg\\gdk-pixbuf\"" "/Fdgdk-pixbuf\gdk-pixbuf-query-loaders.exe.p\queryloaders.c.pdb" /Fogdk-pixbuf/gdk-pixbuf-query-loaders.exe.p/queryloaders.c.obj "/c" ../src/2.42.10-ebb7e46440/gdk-pixbuf/queryloaders.c
cl : Command line warning D9025 : overriding '/W2' with '/W3'
cl : Command line warning D9025 : overriding '/Zi' with '/Z7'
E:\vcpkg\installed\x64-windows\include\glib-2.0\glib/gutils.h(313): warning C4005: 'g_ATEXIT': macro redefinition
E:\vcpkg\installed\x64-windows\include\glibconfig.h(158): note: see previous definition of 'g_ATEXIT'
E:\vcpkg\installed\x64-windows\include\glib-2.0\glib/gutils.h(315): warning C4005: 'g_memmove': macro redefinition
E:\vcpkg\installed\x64-windows\include\glibconfig.h(160): note: see previous definition of 'g_memmove'
../src/2.42.10-ebb7e46440/gdk-pixbuf/queryloaders.c(69): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
../src/2.42.10-ebb7e46440/gdk-pixbuf/queryloaders.c(78): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
../src/2.42.10-ebb7e46440/gdk-pixbuf/queryloaders.c(341): error C2065: 'G_DIR_SEPARATOR': undeclared identifier
../src/2.42.10-ebb7e46440/gdk-pixbuf/queryloaders.c(448): warning C4267: 'initializing': conversion from 'size_t' to 'gint', possible loss of data
...
```
full log: [package-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/11878026/package-x64-windows-dbg-out.log)

This is unexpected because installed glib package defines `G_DIR_SEPARATOR` in `installed\x64-windows\include\glib\gfileutils.h`

**Comment by youknowone** [comment]
Manually adding `#define G_DIR_SEPARATOR '\\'` to the source file was helpful but adding `#include <glibconfig.h>` wasn't.
Maybe something missing to correctly use glibconfig.h.

**Comment by dg0yt** [comment]
> adding #include <glibconfig.h> wasn't.

You were meant to add the location of `glibconfig.h` to the include search path.

**Comment by youknowone** [comment]
@dg0yt I directly added it to the source code.

**Comment by youknowone** [comment]
I had a vcpkg repository building other packages with `x64-windows-static` and first tried to build it as static.
I am trying clean build without -static and maybe it will success. If it success, reproducing the error using mixed triplet of `x64-windows-static` and `x64-windows` will be helpful? or will it be not regarded as bug?

**Comment by dg0yt** [comment]
> `E:\vcpkg\installed\x64-windows\include\glibconfig.h(158): note: see previous definition of 'g_ATEXIT'`

AFAICT you are not using the pristine glib port installation which doesn't have `include/glibconfig.h` but `[debug/]lib/glib-2.0/include/glibconfig.h`. 

So let me rephrase:
Don't move `glibconfig.h` but add its location to the include search path (i.e. `lib/glib-2.0/include` and  `debug/lib/glib-2.0/include`, respectively).

Related: https://github.com/microsoft/vcpkg/issues/32239#issuecomment-1607651549

**Comment by youknowone** [comment]
Thank you so much. I found what's messed up. I had packages from older version of vcpkg.
I am really sorry to bother you all for invalid issue.

Building static + dynamic on clean repository doesn't have any issue.
